File: Anul 2/Medii de Programare si Proiectare/mpp-proiect-csharp-Mothyque/ProiectMPP/ProiectMPP.Server/Networking/RpcConcurrentServer.cs
using ProiectMPP.Common.Networking;
using System;
using System.Net.Sockets;
using System.Threading;

namespace ProiectMPP.Server.Networking
{
    public class RpcConcurrentServer : AbstractServer
    {
        private IServices serverInterface;

        public RpcConcurrentServer(int port, IServices serverInterface)
            : base(port)
        {
            this.serverInterface = serverInterface;
        }

        protected override void ProcessRequest(TcpClient client)
        {
            ClientRpcWorker worker = new ClientRpcWorker(serverInterface, client);

            Thread tw = new Thread(new ThreadStart(worker.Run));
            tw.Start();
        }
    }
}

File: Anul 2/Medii de Programare si Proiectare/Lab/mpp-proiect-java-Mothyque/backend/Server/src/main/java/org/example/networking/RpcConcurrentServer.java
package org.example.networking;

import java.net.Socket;

public class RpcConcurrentServer extends AbstractServer
{
    private org.example.networking.Services serverInterface;

    public RpcConcurrentServer(int port, org.example.networking.Services serverInterface)
    {
        super(port);
        this.serverInterface = serverInterface;
        System.out.println("RpcConcurrentServer created on port" + port);
    }

    @Override
    protected void processRequest(Socket client)
    {
        ClientRpcWorker worker = new ClientRpcWorker(serverInterface, client);
        Thread tw = new Thread(worker);
        tw.start();
    }
}